src/portal/lib/src/config/system/system-settings.component.ts

```diff
@@ -60,9 +95,115 @@ export class SystemSettingsComponent {
         return this.hasAdminRole && this.hasCAFile;
     }
 
-    constructor( @Inject(SERVICE_CONFIG) private configInfo: IServiceConfig) {
+    /**
+     *
+     * Save the changed values
+     *
+     * @memberOf ConfigurationComponent
+     */
+    public save(): void {
+        let changes = this.getChanges();
+        if (!isEmpty(changes)) {
+            this.onGoing = true;
+            this.configService.saveConfigurations(changes)
+                .then(response => {
+                    this.onGoing = false;
+                    // API should return the updated configurations here
+                    // Unfortunately API does not do that
+                    // To refresh the view, we can clone the original data copy
+                    // or force refresh by calling service.
+                    // HERE we choose force way
+                    this.retrieveConfig();
+                    if ('read_only' in changes) {
+                        this.readOnlyChange.emit(changes['read_only']);
```

Comment:
why consider read_only as a specific case?

Reply:
This is from the original code. I did not modified it. I think it because read only will have a global alert